pkg/collect/certificates.go

```diff
@@ -0,0 +1,254 @@
+	RBACErrors
+}
+
+// Certificate collection struct
+type CertCollection struct {
+	Source           *CertificateSource  `json:"source"`
+	Errors           []string            `json:"errors"`
+	CertificateChain []ParsedCertificate `json:"certificateChain"`
+}
+
+// Certificate source
+type CertificateSource struct {
+	SecretName    string `json:"secret,omitempty"`
+	ConfigMapName string `json:"configMap,omitempty"`
+	Namespace     string `json:"namespace,omitempty"`
+}
+
+// Certificate Struct
+type ParsedCertificate struct {
+	CertName                string    `json:"certificate"`
+	Subject                 string    `json:"subject"`
+	SubjectAlternativeNames []string  `json:"subjectAlternativeNames"`
+	Issuer                  string    `json:"issuer"`
+	NotAfter                time.Time `json:"notAfter"`
+	NotBefore               time.Time `json:"notBefore"`
+	IsValid                 bool      `json:"isValid"`
+	IsCA                    bool      `json:"isCA"`
+}
+
+func (c *CollectCertificates) Title() string {
+	return getCollectorName(c)
+}
+
+func (c *CollectCertificates) IsExcluded() (bool, error) {
+	return isExcluded(c.Collector.Exclude)
+}
+
+func (c *CollectCertificates) Collect(progressChan chan<- interface{}) (CollectorResult, error) {
+
+	output := NewResult()
+	results := []CertCollection{}
+
+	// collect certificates from secrets
+	for secretName, namespaces := range c.Collector.Secrets {
+		for _, namespace := range namespaces {
+
+			secretCollections := secretCertCollector(secretName, namespace, c.Client)
+			results = append(results, secretCollections...)
+		}
+	}
+
+	// collect certificates from configMaps
+	for configMapName, namespaces := range c.Collector.ConfigMaps {
+		for _, namespace := range namespaces {
+			configMapCollections := configMapCertCollector(configMapName, namespace, c.Client)
+			results = append(results, configMapCollections...)
+		}
+	}
+
+	certsJson, errCertJson := json.MarshalIndent(results, "", "\t")
+	if errCertJson != nil {
+		return nil, errCertJson
+	}
+
+	filePath := "certificates/certificates.json"
+
+	err := output.SaveResult(c.BundlePath, filePath, bytes.NewBuffer(certsJson))
+	if err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+// configmap certificate collector
+func configMapCertCollector(configMapName string, namespace string, client kubernetes.Interface) []CertCollection {
+
+	results := []CertCollection{}
+	trackErrors := []string{}
+	collection := []ParsedCertificate{}
+
+	getOptions := metav1.GetOptions{}
+
+	// Collect from configMaps
+	configMap, err := client.CoreV1().ConfigMaps(namespace).Get(context.Background(), configMapName, getOptions)
+	if err != nil {
+
+		// collect certificate source information
+		source := &CertificateSource{
+			SecretName: configMapName,
+			Namespace:  namespace,
+		}
+		trackErrors = append(trackErrors, "Either the configMap does not exist in this namespace or RBAC permissions are prenventing certificate collection")
```

Comment:
prenventing -> preventing

Reply:
@banjoh are you aware you can just submit the suggestion for acceptance?
```suggestion
		trackErrors = append(trackErrors, "Either the configMap does not exist in this namespace or RBAC permissions are preventing certificate collection")
```